src-built-in/components/advancedAppCatalog/src/stores/storeActions.js

```diff
@@ -112,60 +111,38 @@ function _addActiveTag(tag) {
  * @param {string} tag The name of the tag
  */
 function _removeActiveTag(tag) {
-
-	let { activeTags, apps } = data;
-
-	let newActiveTags = activeTags.filter((currentTag) => {
+	data.activeTags = data.activeTags.filter((currentTag) => {
 		return currentTag !== tag;
 	});
 
-	let newApps = apps.filter((app) => {
-		for (let i = 0; i < newActiveTags.length; i++) {
-			let tag = activeTags[i].trim();
-			if (app.tags.includes(tag)) {
-				return true;
-			}
-		}
-	});
-
-	getStore().setValues([{
-		field: "activeTags",
-		value: newActiveTags
-	}, {
-		field: "filteredApps",
-		value: newApps
-	}]);
+	getStore().setValue({ field: "activeTags", value: data.activeTags	});
+	filterApps();
 }
 
 function _refreshTags() {
-	let { activeTags, apps } = data;
-
-	let newApps = apps.filter((app) => {
-		for (let i = 0; i < activeTags.length; i++) {
-			const tag = activeTags[i].trim();
-			if (app.tags.includes(tag)) {
-				return true;
-			}
-		}
-	});
-
-	getStore().setValues([{
-		field: 'activeTags',
-		value: activeTags
-	}, {
-		field: 'filteredApps',
-		value: newApps
-	}]);
+	filterApps();
 }
 
 /**
  * Clears all active tags
  */
 function _clearActiveTags() {
-	getStore().setValue({
-		field: "activeTags",
-		value: []
-	});
+	getStore().setValue({ field: "activeTags", value: [] });	
+}
+
+
+function filterApps() {
+	let { activeTags, searchText } = data;
+	
+
+	if (activeTags.length === 0 && searchText === "") {
+		goHome();
+	} else {
+		appd.search({ text: searchText, tags: activeTags }, (err, data) => {
+			if (err) console.log("Failed to search apps");
```

Comment:
So, there was an error but we're going to continue to work with `data` that could be unusable information?

I'd expect a different exit pathway for error handling compared to continuing to work. If we can continue operations just fine, why log this at all?

Even if we do need this log, it should probably fall towards a warning. And also be logged to the Central Logger and not the DevTools console.

Reply:
This was how it was previously written. However, I agree. Will log to central logger as an error and return.